src/org/zaproxy/zap/extension/dynssl/ExtensionDynSSL.java

```diff
@@ -146,4 +153,54 @@ public Certificate getRootCA() throws KeyStoreException {
 	public boolean supportsDb(String type) {
     	return true;
     }
+	
+	/**
+	 * Gets ZAPs current CA Certificate in X.509 format.
+	 * 
+	 * @return The X.509 version of ZAPs current CA certificate.
+	 */
+	public X509Certificate getCaCertificate() {
+		try {
+			return (X509Certificate) getRootCA();
+		} catch (KeyStoreException e) {
+			logger.error("Couldn't get ZAP's Root CA Certificate", e);
+			return null;
+		}
+	}
+	
+	/**
+	 * Returns true if the certificate expired before the current date, otherwise false.
+	 * 
+	 * @param cert the X.509 certificate for which expiration should be checked.
+	 * @return true if the certificate has expired, otherwise false.
+	 */
+	public boolean isCertExpired(X509Certificate cert) {
+		if (cert != null) {
+			if (cert.getNotAfter().before(new Date())) {
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	/**
+	 * Displays a warning dialog, and logs a warning message if the certificate has expired.
+	 * (Assumes the certificate being passed in has expired.)
+	 * 
+	 * @param cert the X.509 certification which is being warned about.
+	 * @see #isCertExpired(X509Certificate)
+	 */
+	public void warnCertExpired(X509Certificate cert) {
+		if (cert == null) {
+			return;
+		}
+		StringBuilder warnMsgSB = new StringBuilder();
+		warnMsgSB.append(Constant.messages.getString("dynssl.warn.cert.expired", cert.getNotAfter().toString(),
+				new Date().toString()));
+		warnMsgSB.append('\n');
+    	if (View.isInitialised() && cert != null) {

```

Comment:
- The null check can be removed (already done earlier).
 - Mixed indentation chars.